### PR TITLE
Akamai Guardicore Event Collector

### PR DESCRIPTION
**Pull request by dansterenson**
# Note
* Still need to add UT.
* Authentication functions were taken from Guardicore V2 integration.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue
](https://jira-hq.paloaltonetworks.local/browse/CIAC-6559?filter=-1)

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    end_time = int(datetime.now().timestamp() * 1000)
+    demisto.debug(f'Getting events from: {timestamp_to_datestring(start_time)}, till: {timestamp_to_datestring(end_time)}')
+    offset = arg_to_number(last_run.get('offset')) or 0
+    limit = arg_to_number(params.get('max_events_per_fetch')) or 1000
+
+    retrieve_events = client.get_events(start_time, end_time, limit, offset)
+    demisto.debug(f'Fetched {len(retrieve_events)} events.')
+
+    demisto.setLastRun(
+        {'from_ts': end_time} if len(retrieve_events) < limit else {'from_ts': start_time, 'offset': offset + limit}
```
Comment (DeanArbel) [inline Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py:198]: Does the API treat `start_time` as `gt` or `gte`? i.e. will there be a need to dedupe the data between 2 subsequent runs?

File: Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+    Args:
+        events: list[dict] - list of events to add the _time key to.
+
+    Returns:
+        list: The events with the _time key.
+    """
+    if events:
+        for event in events:
+            event['_time'] = event.get('start_time')
```
Comment (DeanArbel) [inline Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py:166]: It should be faster to do this in comprehension i.e.
```suggestion
        [{**event, '_time': event.get('start_time')} for event in events]
```

File: Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+       Gets events from Guardicore API.
+    """
+    start_time = date_to_timestamp(
+        arg_to_datetime(args.get('from_date', '1 second ago')))
+    end_time = date_to_timestamp(
+        arg_to_datetime(args.get('to_time'))) if "to_time" in args else int(datetime.now().timestamp() * 1000)
+    limit = arg_to_number(args.get("limit", 1000))
+    offset = int(args.get('offset', 0))
+    events = client.get_events(start_time, end_time, limit, offset)
+    hr = tableToMarkdown(name='Test Event', t=events)
```
Comment (DeanArbel) [inline Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py:180]: These aren't necessarily test events

File: Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+category: Analytics & SIEM
+ectionOrder:
+- Connect
+- Collect
+commonfields:
+  id: AkamaiGuardicoreEventCollector
+  version: -1
+configuration:
+- defaultvalue: https://some_url.com
```
Comment (DeanArbel) [inline Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.yml:9]: Is there some other default URL that can be configured?

File: Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector_description.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+To configure the integration, enter your credentials in the following fields:
+- Username
+- Password
+
+### To configure fetch events
+- Check the *Fetches events*.
+- Set *Max events number per fetch* the the fetch events parameters \9default 1000)
```
Comment (DeanArbel) [inline Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector_description.md:7]:
```suggestion
- Set *Max events number per fetch* the the fetch events parameters. default 1000)
```

**Review by DeanArbel — CHANGES_REQUESTED** [review]

**Review by DeanArbel — APPROVED** [review]
lgtm

**Review by eepstain — APPROVED** [review]

**Pull request closed without merging**